src/pages/slots/forms/EditCapacityForm.tsx

```diff
@@ -0,0 +1,50 @@
+import { FormContainer, Input, Label, InputContainer, Button } from "./EditCapacityForm.styles";
+import * as yup from "yup";
+import { EditCapacityScheme } from "./EditCapacity.scheme";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useForm } from "react-hook-form";
+import { ErrorMessage } from "../../../components/error_message/ErrorMessage";
+import { forwardRef, useImperativeHandle } from 'react';
+
+export interface EditSlotFormHandle {
+    submitForm: () => void;
+}
+
+
+const EditSlotForm = forwardRef<EditSlotFormHandle>((props, ref) => {
+
+
+
+    type FormData = yup.InferType<typeof EditCapacityScheme>;
+
+    const {
+        register,
+        handleSubmit,
+        formState: { errors },
+    } = useForm<FormData>({
+        resolver: yupResolver(EditCapacityScheme),
+    });
+
+    const onSubmit = (data: FormData) => {
+        console.log(data);
+    };
+
+    useImperativeHandle(ref, () => ({
+        submitForm: () => {
+            handleSubmit(onSubmit)();
+        }
+    }));
```

Comment:
Esto, ahora cambia de esta manera


```suggestion
    useImperativeHandle(ref, () => ({
        submitForm: () =>
            new Promise<boolean>((resolve) => {
                handleSubmit(
                    async (data) => {
                        try {
                            await onSubmit(data);
                            resolve(true);
                        } catch (e) {
                            resolve(false);
                        }
                    },
                    () => {
                        resolve(false);
                    }
                )();
            }),
    }));
```

Basicamente lo que hacemos es devolver una promesa que dice "se completo correctamente el proceso o no", entonces esperamos la ejecucion del `onSubmit(data)` y si todo salio bien, "resolvemos" con un true (todo salio ok), caso contrario retornamos un false. Este true/false es lo que devolvemos y leemos en el `CreateSlot.tsx` para cerrar el modal